sympy/core/power.py

```diff
@@ -1582,8 +1583,8 @@ def mul(d1, d2):
             # Convert floats like 0.5 to exact SymPy numbers like S.Half, to
             # prevent rounding errors which can induce wrong values of d leading
             # to a NotImplementedError being returned from the block below.
-            from sympy.simplify.simplify import nsimplify
-            _, d = nsimplify(g).leadterm(x, logx=logx)
+            g.replace(lambda x: x.is_Float and x.is_same(Rational(x), equal_valued), lambda x: Rational(x))
+            _, d = g.leadterm(x, logx=logx)
```

Comment:
I personally don't prefer having these huge conditional checks :)
 Always try to look for something more fundamental that could address this. Is this the best we can do ?

Reply:
What I think about this process is that fundamental problem for introduction of the following conditional(line1581-1585) was to handle correctly the removal of the constant term as mentioned before https://github.com/sympy/sympy/blob/36bcecfeb5b359238a6047c7888555e196d43c89/sympy/core/power.py#L1581-L1585
what I think is that the other changes I have made satisfactorily helps us in that part and the above conditional seemed not that helpful to me anymore as it would most probably not hit this conditional anymore ,though I was wary whether we should be doing away with this conditional as there may always be a possibility of resurgence of similar issue if someone comes with a fringe case that breaks this, `nsimplify` in that regards could have been helpfull (as was previously) but given its limitation to certain digits after decimal(thanks to this issue which pointe d this out),replacing the nsimplify may be a better way out here. This change that I incorporated as suggested by @smichr, in my opinion will suffice in this matter given the fact that scope of hitting this code has already been reduced by the other changes made.

Reply:
You still have a constant term whether you use Float or Rational.